src/MessageStore.h

```diff
@@ -35,19 +35,17 @@
 
 namespace arangodb { namespace fuerte { inline namespace v1 {
 // MessageStore keeps a thread safe list of all requests that are "in-flight".
-template <class RequestItemT>
+template <typename RequestItemT>
 class MessageStore {
  public:
   // add a given item to the store (indexed by its ID).
   void add(std::shared_ptr<RequestItemT> item) {
-    std::lock_guard<std::mutex> lockMap(_mutex);
```

Comment:
as this mutex is removed here entirely, I hope there is now some other external synchronization for accessing the `MessageStore`.

Reply:
So from my understanding of @graetzer's explanation, the `MessageStore` is only accessed by the single `asio` thread, and therefore does not need to be synchronized. All the locking code had previously been commented out for the better part of the last year in arangod, and I just removed it when porting changes.

Reply:
Ok, alright then.